spec/controllers/api/enterprise_logo_controller_spec.rb

```diff
@@ -0,0 +1,92 @@
+require "spec_helper"
+
+module Api
+  describe EnterpriseLogoController, type: :controller do
+    include AuthenticationWorkflow
+
+    let(:admin_user) { create(:admin_user) }
+    let(:enterprise_owner) { create(:user) }
+    let(:enterprise) { create(:enterprise, owner: enterprise_owner ) }
+    let(:enterprise_manager) { create(:user, enterprise_limit: 10, enterprises: [enterprise]) }
+    let(:other_enterprise_owner) { create(:user) }
+    let(:other_enterprise) { create(:enterprise, owner: other_enterprise_owner ) }
+    let(:other_enterprise_manager) { create(:user, enterprise_limit: 10, enterprises: [other_enterprise]) }
+
+    describe "removing logo" do
+      image_path = File.open(Rails.root.join("app", "assets", "images", "logo-black.png"))
+      let(:image) { Rack::Test::UploadedFile.new(image_path, "image/png") }
+
+      let(:enterprise) { create(:enterprise, owner: enterprise_owner, logo: image) }
+
+      before do
+        allow(controller).to receive(:spree_current_user) { current_user }
+      end
+
+      context "as manager" do
+        let(:current_user) { enterprise_manager }
+
+        it "removes logo" do
+          expect(enterprise.logo?).to be true
```

Comment:
I'm not sure this check is worth the added test suite execution time. We should trust the state we set up in line 19 and if that fails, we will notice way before seeing this test fail, IMO.

Reply:
Hmm... You're right, I agree.